astropy/io/votable/tree.py

```diff
@@ -264,6 +264,32 @@ def _get_unit_format(config):
     return format
 
 
+def _attach_default_config(votable, config) -> dict:
+    """Create default config if none given.
+
+    Parameters
+    ----------
+    votable
+        Object that might potentially have a config attached to it.
+    config : `dict` or `None`
+        Config to be used or defined.
+
+    Returns
+    -------
+    config : `dict`
+        If config was `None` this will be a new `dict`. If the ``votable``
+        supported version checks then version information will be stored in
+        the returned `dict`. If it was defined this will be the given
+        parameter unchanged.
+    """
+    if config is None:
+        if hasattr(votable, "_get_version_checks"):
```

Comment:
Now that I've made it so that `VOTableFile` always has the version checks set for it, it's probably better to make this line instead be:

```suggestion
        if hasattr(votable, "config"):
```
and then copy the entire config over.

Reply:
That's a good point.  I don't have a strong preference, though limiting the scope to the version configs makes it a little more explicit why this is being done and is closest to what was being done before with `Field`.  Copying the whole config could bring in config items from the votable, which is probably not harmful and might even be good in most cases, but it does go beyond what is explicitly needed for this fix. 

It's messy that version concerns aren't better encapsulated (away from configs).  Without bigger refactoring, it will be hard to make configs really safe during programmatic construction of a votable, i.e., that they meet all the same implicit assumptions as when they are created during parsing.  The API allows passing anything in for config in each element constructor.

Reply:
Ok. I'll leave it with the explicit `_get_version_checks` call, although I will add "version" field into the dict so that all the version information is present.